cmd/traffic_cache_tool/CacheScan.cc

```diff
@@ -38,13 +38,21 @@ CacheScan::Scan()
 {
   int64_t guessed_size = 1048576; // 1M
   Errata zret;
+  std::bitset<65536> dir_bitset;
   char *stripe_buff2 = (char *)ats_memalign(ats_pagesize(), guessed_size);
   for (int s = 0; s < this->stripe->_segments; s++) {
+    dir_bitset.reset();
     for (int b = 0; b < this->stripe->_buckets; b++) {
       CacheDirEntry *seg = this->stripe->dir_segment(s);
       CacheDirEntry *e   = dir_bucket(b, seg);
       if (dir_offset(e)) {
         do {
+          if (dir_bitset[dir_to_offset(e, seg)]) {
```

Comment:
`dir_to_offset` - is that really bounded by 64K?

Reply:
yes.. it basically returns a 16bit integer (type uint16_t)